src/webgpu/api/validation/encoding/render_bundle.spec.ts

```diff
@@ -51,3 +59,239 @@ g.test('render_bundles,device_mismatch')
+
+    const compatible = bundleFormat === passFormat;
+
+    const bundleEncoder = t.device.createRenderBundleEncoder({
+      colorFormats: [],
+      depthStencilFormat: bundleFormat as GPUTextureFormat,
+    });
+    const bundle = bundleEncoder.finish();
+
+    const encoder = t.createEncoder('render pass', {
+      attachmentInfo: {
+        colorFormats: [],
+        depthStencilFormat: passFormat as GPUTextureFormat,
+      },
+    });
+    encoder.encoder.executeBundles([bundle]);
+
+    encoder.validateFinish(compatible);
+  });
+
+g.test('depth_stencil_readonly_mismatch')
+  .desc(
+    `
+    Tests executeBundles cannot be called with render bundles that do match the depthStencil
+    readonly state of the render pass.
+    `
+  )
+  .params(u =>
+    u.combineWithParams([
+      {
+        bundleDepthReadOnly: false,
+        bundleStencilReadOnly: false,
+        passDepthReadOnly: false,
+        passStencilReadOnly: false,
+      },
+      {
+        bundleDepthReadOnly: true,
+        bundleStencilReadOnly: false,
+        passDepthReadOnly: true,
+        passStencilReadOnly: false,
+      },
+      {
+        bundleDepthReadOnly: false,
+        bundleStencilReadOnly: true,
+        passDepthReadOnly: false,
+        passStencilReadOnly: true,
+      },
+      {
+        bundleDepthReadOnly: true,
+        bundleStencilReadOnly: true,
+        passDepthReadOnly: true,
+        passStencilReadOnly: true,
+      },
+      {
+        bundleDepthReadOnly: true,
+        bundleStencilReadOnly: false,
+        passDepthReadOnly: true,
+        passStencilReadOnly: true,
+      },
+      {
+        bundleDepthReadOnly: false,
+        bundleStencilReadOnly: true,
+        passDepthReadOnly: true,
+        passStencilReadOnly: true,
+      },
+      {
+        bundleDepthReadOnly: false,
+        bundleStencilReadOnly: false,
+        passDepthReadOnly: true,
+        passStencilReadOnly: true,
+      },
+      {
+        bundleDepthReadOnly: true,
+        bundleStencilReadOnly: true,
+        passDepthReadOnly: false,
+        passStencilReadOnly: true,
+      },
+      {
+        bundleDepthReadOnly: true,
+        bundleStencilReadOnly: true,
+        passDepthReadOnly: true,
+        passStencilReadOnly: false,
+      },
+      {
+        bundleDepthReadOnly: true,
+        bundleStencilReadOnly: true,
+        passDepthReadOnly: false,
+        passStencilReadOnly: false,
+      },
+    ])
+  )
+  .fn(async t => {
+    const {
+      bundleDepthReadOnly,
+      bundleStencilReadOnly,
+      passDepthReadOnly,
+      passStencilReadOnly,
+    } = t.params;
+
+    const compatible =
+      bundleDepthReadOnly === passDepthReadOnly && bundleStencilReadOnly === passStencilReadOnly;
+
+    const bundleEncoder = t.device.createRenderBundleEncoder({
+      colorFormats: [],
+      depthStencilFormat: 'depth24plus',
+      depthReadOnly: bundleDepthReadOnly,
+      stencilReadOnly: bundleStencilReadOnly,
```

Comment:
This test should run with all of the depth-stencil formats.

Something that's interesting in the spec is that the render bundle encoder directly inherits readOnly from the descriptor.

i.e. `encoder.[[depthReadOnly]] = descriptor.depthReadOnly`

However, maybe `encoder.[[depthReadOnly]] should = descriptor.depthReadOnly || formatHasNoDepth(fmt)` ?
same for passes.

In any case, this test LGTM w.r.t. the current spec.